src/boot/url-template.js

```diff
@@ -52,6 +45,10 @@ export default function processUrlTemplate(url, document_ = document) {
   if (origin) {
     url = url.replace('{current_host}', origin.hostname);
     url = url.replace('{current_scheme}', origin.protocol);
+  } else {
+    throw new Error(
+      'Could not process URL template because script origin is unknown'
+    );
```

Comment:
There is not currently a situation where we would expect this to be triggered during client startup. If it does happen for an unforeseen reason, this error will provide a clear pointer to the problem.